cpu/nrf5x_common/periph/i2c_nrf52_nrf9160.c

```diff
@@ -126,6 +126,15 @@ static int finish(i2c_t dev, int inten_success_flag)
         DEBUG("[i2c] finish: stop event occurred\n");
     }
 
+    if (inten_success_flag & TWIM_INTEN_LASTTX_Msk) {
+        /* The interrupt is raised already when the last TX is started, but we
+         * have to wait until it was actually transmitted lest the transmission
+         * would be suppressed immediately by the next following write --
+         * careful here: enabling DEBUG introduces enough latency that the
+         * issue doesn't show up any more. */
+        while (bus(dev)->TXD.AMOUNT != bus(dev)->TXD.MAXCNT) {}
```

Comment:
Should this also check `bus(dev)->EVENTS_ERROR`?

Reply:
Good catch, thanks -- indeed.

Reply:
Fixed with the last fixup. I don't have any means of testing whether this works: I'd need an I2C peer that I can fault precisely between LASTTX and the completion.

Reply:
Did you push?

Reply:
Yes but to the wrong branch (pushed it to #20303 where I wrote it because that's based on this and there I had better chances of testing it).